Python/pylifecycle.c

```diff
@@ -2897,6 +2897,30 @@ fatal_error_exit(int status)
     }
 }
 
+static inline int
+acquire_dict_lock_for_dump(PyObject *obj) {
+#ifdef Py_GIL_DISABLED
+    PyMutex *mutex = &obj->ob_mutex;
+    if (_PyMutex_LockTimed(mutex, 0, 0) == PY_LOCK_ACQUIRED) {
+        return 0;
+    }
+    return -1;
+#else
+    return 0;
+#endif
+}
+
+static inline void
+release_dict_lock_for_dump(PyObject *obj) {
+#ifdef Py_GIL_DISABLED
+    PyMutex *mutex = &obj->ob_mutex;
+    uint8_t expected = _Py_LOCKED;
+    // Do not wake up other threads.
```

Comment:
Why not using PyMutex_Unlock()?

Reply:
https://github.com/python/cpython/issues/120837#issue-2366781894

```
we can't call PyMutex_Unlock() because it's not async-signal-safe (it internally acquires locks in order to wake up threads), so we should either use a simple atomic exchange to unlock the dict (without waking up waiters) or not bother unlocking the lock at all. We exit shortly after _Py_DumpExtensionModules, so it doesn't matter if we don't wake up other threads.
```

Reply:
Please add a comment to explain shortly why you don't simply call PyMutex_Unlock().

Reply:
Done!